gateway/src/apicast/policy/maintenance_mode/maintenance_mode.lua

```diff
@@ -0,0 +1,34 @@
+-- This is a simple policy. It allows you reject incoming requests with a
+-- specified status code and message.
+-- It's useful for maintenance periods or to temporarily block an API
+
+local _M = require('apicast.policy').new('Maintenance mode', 'builtin')
+
+local tonumber = tonumber
+local new = _M.new
+
+local default_status_code = 503
+local default_message = "Service Unavailable - Maintenance"
+
+function _M.new(configuration)
+  local policy = new(configuration)
+
+  policy.status_code = default_status_code
+  policy.message = default_message
+
+  if configuration then
+    policy.status_code = tonumber(configuration.status) or policy.status_code
+    policy.message = configuration.message or policy.message
```

Comment:
It might be useful to also set some header in case this is active. So you can have the payload as JSON for example.

Reply:
Not sure what you mean by setting a header here.
Do you mean returning JSON or plain text according to the value of the `Accept` header of the request?

Reply:
I mean setting also Content-Type header when the policy is active. JSON payload is incorrect without proper Content-Type. And using two policies to activate mantenance mode would be weird.

Reply:
:+1:

https://github.com/3scale/APIcast/pull/1105/commits/bcee0a9400b1604fc90ceb96a05df61f59165b0f